modules/alloydb/variables.tf

```
@@ -245,21 +245,33 @@ variable "name" {
 }
 
 variable "network_config" {
-  description = "Network configuration for cluster and instance. Only one between cluster_network_config and cluster_psc_config can be used."
+  description = "Network configuration for cluster and instance. Only one between psa_config and psc_config can be used."
   type = object({
-    network                      = string
-    allocated_ip_range           = optional(string, null)
-    authorized_external_networks = optional(list(string), null)
-    enable_public_ip             = optional(bool, false)
+    psa_config = optional(object({
+      network                      = optional(string, null)
+      allocated_ip_range           = optional(string, null)
+      authorized_external_networks = optional(list(string), null)
+      enable_public_ip             = optional(bool, false)
+    }), null)
+    psc_config = optional(object({
+      allowed_consumer_projects = optional(list(string), [])
+    }), null)
   })
   nullable = false
   validation {
     condition = (
-      (try(length(var.network_config.authorized_external_networks), 0) != 0 && var.network_config.enable_public_ip)
-      || try(length(var.network_config.authorized_external_networks), 0) == 0
+      var.network_config.psa_config == null || (
+        (try(var.network_config.psa_config.enable_public_ip, false) &&
+        try(length(var.network_config.psa_config.authorized_external_networks), 0) > 0) ||
+        try(length(var.network_config.authorized_external_networks), 0) == 0
```

```suggestion
        try(length(var.network_config.psa_config.authorized_external_networks), 0) == 0
```
This was the intention?